### PR TITLE
libuv: Forward additional getaddrinfo errors

### PR DESCRIPTION
**Pull request by skabbes**
These are defined on android and on mac (and possibly other platforms)

### CODE REVIEW

File: src/uv-common.c [hunk 1]
```diff
@@ -433,6 +433,18 @@ int uv__getaddrinfo_translate_error(int sys_err) {
...
+  case EAI_BADHINTS: return UV_EAI_BADHINTS;
+#endif
+#if defined(EAI_PROTOCOL)
+  case EAI_PROTOCOL: return UV_EAI_PROTOCOL;
+#endif
+#if defined(EAI_OVERFLOW)
+  case EAI_OVERFLOW: return UV_EAI_OVERFLOW;
+#endif
+#if defined(EAI_MAX)
+  case EAI_MAX: return UV_EAI_MAX;
```
Comment (bnoordhuis) [inline src/uv-common.c]: Is EAI_MAX actually returned or is it a 'highest error code plus one' marker?

**Comment by skabbes** [comment]
I'm pretty sure that its only 'highest' plus one.  I didn't know what the libuv philosophy was for that kind of thing.  I also wanted to allow the user to be able to avoid [this abort](https://github.com/joyent/libuv/blob/master/src/uv-common.c#L437-L440)

I can remove it if you like

**Comment by bnoordhuis** [comment]
> I can remove it if you like

Yes, please.

**Comment by skabbes** [comment]
fixed, but i have no idea why the smartos and windows tests are failing now

**Comment by bnoordhuis** [comment]
Looks unrelated.  Landed in dbe9905.  Thanks, Steven.

**Pull request closed without merging**